dataactbroker/handlers/fileHandler.py

```diff
@@ -505,20 +510,89 @@ def startGenerationJob(self, submission_id, file_type):
                 job.end_date = datetime.strptime(end_date,"%m/%d/%Y").date()
                 valJob.start_date = datetime.strptime(start_date,"%m/%d/%Y").date()
                 valJob.end_date = datetime.strptime(end_date,"%m/%d/%Y").date()
+                # Generate random uuid and store generation task
+                task_key = uuid4()
+                task = FileGenerationTask(generation_task_key = task_key, submission_id = submission_id, file_type_id = jobDb.getFileTypeId(file_type_name))
+                jobDb.session.add(task)
+
                 jobDb.session.commit()
             except ValueError as e:
                 # Date was not in expected format
                 exc = ResponseException(str(e),StatusCode.CLIENT_ERROR,ValueError)
                 return False, JsonResponse.error(exc, exc.status, url = "", start = "", end = "",  file_type = file_type)
-            get_url = CONFIG_BROKER["".join([file_type_name, "_url"])].format(cgac_code, start_date, end_date)
-            jq.generate_d_file.delay(get_url, user_id, job.job_id, InterfaceHolder, timestamped_name, self.isLocal)
+            # Create file D API URL with dates and callback URL
+            callback = "https://{}:{}/v1/complete_generation/{}/".format(CONFIG_SERVICES["broker_api_host"], CONFIG_SERVICES["broker_api_port"],task_key)
+            get_url = CONFIG_BROKER["".join([file_type_name, "_url"])].format(cgac_code, start_date, end_date, callback)
```

Comment:
This is a small thing, and I can see it both ways, but it's hard to know what's going on here. Due to the combination of `.join` and `format` to construct `get_url`, it took me a while to realize that the `format` operating on parameters in the config file.

Does it make sense for readability to keep the API endpoints in the config file and add the parameters here? I.e., config.yml just has `https://data-usas.uat.workplace.gov/api/contracts`.

Honestly not sure what the best practice is here.